### ISSUE TITLE
use-of-default-aes detects false positive when calling KeyGenerator.getInstance("AES")

### ISSUE THREAD

**Issue by creste** [returntocorp/sgrep-rules#2796]
**Describe the bug**
[use-of-default-aes](https://github.com/returntocorp/semgrep-rules/blob/13889a6755be807ae133e0f798654ac14ca2ef98/java/lang/security/audit/crypto/use-of-default-aes.yaml) flags [KeyGenerator.getInstance("AES")](https://docs.oracle.com/javase/7/docs/api/javax/crypto/KeyGenerator.html#getInstance(java.lang.String)) as a false positive.  Per the KeyGenerator API documentation, the only way to create a new key generator for AES is to pass in the string literal `"AES"`.

**To Reproduce**
Scan this code with the [use-of-default-aes](https://github.com/returntocorp/semgrep-rules/blob/13889a6755be807ae133e0f798654ac14ca2ef98/java/lang/security/audit/crypto/use-of-default-aes.yaml) rule:
```java
var keyGenerator = KeyGenerator.getInstance("AES");
```

**Expected behavior**
No findings are reported.

**Priority**
How important is this to you?
- [ ] P0: blocking me from making progress
- [ ] P1: this will block me in the near future
- [X] P2: annoying but not blocking me

**Comment by enncoded** [comment]
https://github.com/returntocorp/semgrep-rules/pull/2803 fixes it
